src/components/SignUp.jsx

```diff
@@ -1,20 +1,30 @@
-import {
-  Box,
-  Button,
-  FormControl,
-  InputLabel,
-  MenuItem,
-  Select,
-  Stack,
-  TextField,
-  Typography,
-} from "@mui/material";
+import {Box,Button,FormControl,InputLabel,MenuItem,Select,Stack,TextField,Typography,} from "@mui/material";
 import { useFormik } from "formik";
-import React from "react";
-import { NavLink } from "react-router-dom";
+import React, { useEffect } from "react";
+import { NavLink, useNavigate } from "react-router-dom";
 import { signUpSchema } from "../schemas";
-import "../components/stylecss/style.css"
+import "../components/stylecss/style.css";
+import { useDispatch, useSelector } from "react-redux";
+import { resetReducer, signUpApi } from "../redux/Slices/signupSlice";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
 const SignUp = () => {
+  const navigate = useNavigate();
+
+  const dispatch = useDispatch();
```

Comment:
why you are using dispatch from this function if you are already exported from store